.github/workflows/release.yml

```diff
@@ -0,0 +1,340 @@
+name: Release
```

Comment:
:x: Please add a concurrency config like we do for [deploying snapshots](https://github.com/camunda/zeebe/blob/38d8edf95746f4060d725ad5771ed89f17db2a34/.github/workflows/ci.yml#L494-L496). This ensures that we don't produce a mess when two releases are triggered concurrently.

Reply:
wdyt about doing this based on the release-branch used? that would still allow e.g. doing multiple patch releases in parallel which seems reasonable to do?

Reply:
Yeah that seems reasonable :+1:

Reply:
https://github.com/camunda/zeebe/pull/11240/commits/1d33d49699eb747aa67b56910117cdb6ddee532c